pumps/kafka.go

```diff
@@ -61,11 +70,33 @@ func (k *KafkaPump) Init(config interface{}) error {
 		}
 	}
 
+	var mechanism sasl.Mechanism
+
+	switch k.kafkaConf.SASLMechanism {
+	case "":
+		break
+	case "PLAIN", "plain":
+		mechanism = plain.Mechanism{Username: k.kafkaConf.Username, Password: k.kafkaConf.Password}
+	case "SCRAM", "scram":
+		algorithm := scram.SHA256
+		if k.kafkaConf.Algorithm == "sha-512" || k.kafkaConf.Algorithm == "SHA-512" {
+			algorithm = scram.SHA512
+		}
+		var mechErr error
+		mechanism, mechErr = scram.Mechanism(algorithm, k.kafkaConf.Username, k.kafkaConf.Password)
+		if mechErr != nil {
+			k.log.Fatal("Failed initialize kafka mechanism  : ", mechErr)
+		}
+	default:
+		k.log.Warn("Tyk pump doesn't support " + k.kafkaConf.SASLMechanism + " mechanism.")
```

Comment:
Can we please add the word "sasl" to the message? for instance:
`k.log.WithField("SASL-Mechanism", k.kafkaConf.SASLMechanism).Warn("Tyk pump doesn't support this SASL mechanism"")`

Reply:
Why not supporting sasl? That is the main auth type used in enterprise clients.

Reply:
It's going to default only when the SASL Mechanism isn't **plain** or **scram**.

Reply:
But default is actually the main need for enterprise clients and it looks like it is not supported. Maybe something interesting here https://github.com/confluentinc/confluent-kafka-go ?

Reply:
@bmonteiro based on your example current PR covers your scenario. Just set:
```
"sasl_mechanism": "plain",
"sasl_username": "token",
"sasl_password": "api-key-secret"
```